### DISCUSSION TITLE
How to disable formatting on paste?

### DISCUSSION THREAD

**Discussion by adam-nielsen** [ananthakumaran/tide#470] [Q&A]
When I paste/yank text into a .tsx file, something runs to prettify the whole file, but it takes around 30 seconds to do so and often undoes the formatting that I want.  If I press Ctrl+G a few times to cancel it, it ends up completely corrupting the buffer so rather than sit through it every time, I would like to disable it completely and just get a normal paste like every other mode offers.

Is there a way to disable this format-on-paste feature?

I have `'tide-format-before-save` set and this fixes up the formatting when I save the file, so I don't need this extra level of formatting when I paste, which often formats the file incorrectly as well as being incredibly slow.

**Comment by ananthakumaran** [comment]
> Is there a way to disable this format-on-paste feature?

I don't think tide does it. To rule out `tide-format-before-save` function, run `M-x trace-function` and type tide-format-before-save. Each call to it will get logged. 

`M-x toggle-debug-on-quit` can be used to get the stack trace of the current command. Paste some code and press <kbd>Ctrl + G</kbd> and you should get a stack trace which might give some hint.
